packages/colorpickers/src/elements/ColorDialog/index.tsx

```diff
@@ -0,0 +1,199 @@
+ */
+
+import React, {
+  useRef,
+  useState,
+  useEffect,
+  useContext,
+  Children,
+  cloneElement,
+  forwardRef,
+  ReactElement,
+  HTMLAttributes
+} from 'react';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { ThemeContext } from 'styled-components';
+import { Modifier } from 'react-popper';
+import { Button } from '@zendeskgarden/react-buttons';
+import { GARDEN_PLACEMENT } from '@zendeskgarden/react-modals';
+import Chevron from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';
+import { ColorPicker, IColorPickerProps } from '../ColorPicker';
+import { IColor } from '../../utils/types';
+import {
+  StyledButton,
+  StyledDialogPreview,
+  StyledPreviewContainer,
+  StyledCheckered,
+  StyledTooltipModal,
+  StyledTooltipBody
+} from '../../styled';
+
+export interface IColorDialogProps extends IColorPickerProps {
+  /**
+   * Handles close actions. Can be triggered from the backdrop.
+   *
+   * @param {Object} color A color picker state
+   */
+  onClose?: (color: IColor) => void;
+  /** Adjusts the placement of the color dialog */
+  placement?: GARDEN_PLACEMENT;
+  /** Disables the color dialog button */
+  disabled?: boolean;
+  /**
+   * Modifies [Popper instance](https://popper.js.org/docs/v2/modifiers/) to customize positioning logic
+   */
+  popperModifiers?: Partial<Modifier<any, any>>[];
+  /**
+   * Sets the `z-index` of the color dialog
+   */
+  zIndex?: number;
+  /**
+   * Adds an arrow to the color dialog
+   */
+  hasArrow?: boolean;
+  /**
+   * Animates the color dialog
+   */
+  isAnimated?: boolean;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLButtonElement>
+ */
+export const ColorDialog = forwardRef<
+  HTMLButtonElement,
+  IColorDialogProps & Omit<HTMLAttributes<HTMLButtonElement>, 'color' | 'onChange'>
+>(
+  (
+    {
+      color,
+      defaultColor,
+      placement,
+      onChange,
+      onClose,
+      labels,
+      hasArrow,
+      isAnimated,
+      popperModifiers,
+      zIndex,
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    const theme = useContext(ThemeContext);
+    const isControlled = color !== null && color !== undefined;
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const colorPickerRef = useRef<HTMLDivElement>(null);
+    const mergedRef = mergeRefs([ref, buttonRef]);
+    const [isOpen, setIsOpen] = useState(false);
+    const [referenceElement, setReferenceElement] = useState<HTMLButtonElement | null>();
+    const [uncontrolledColor, setUncontrolledColor] = useState<string | IColor | undefined>(
+      defaultColor
+    );
+
+    useEffect(() => {
+      let timeout: number;
+
+      if (isOpen) {
+        setReferenceElement(buttonRef.current);
+      } else if (isAnimated) {
+        // Match the duration of the menu fade out transition.
+        timeout = window.setTimeout(() => setReferenceElement(null), 200);
+      } else {
+        setReferenceElement(null);
+      }
```

Comment:
@hzhu I believe this should live on the `TooltipModal` so that it is fixed for all tooltip modal instances.

Reply:
Do you think that would belong in a new bug fix pull request for `react-modals`? If so, I can remove it here and open a separate PR.

Reply:
I'm fine with that being a separate PR. We'll just want to be sure the fix works here as expected.